src/platform/qt/scripting/ScriptingTextBuffer.cpp

```diff
@@ -49,26 +49,45 @@ void ScriptingTextBuffer::setBufferName(const QString& name) {
 	emit bufferNameChanged(name);
 }
 
+void ScriptingTextBuffer::lineBreak() {
+	bool nextBlockExists = m_shim.cursor.movePosition(QTextCursor::NextBlock, QTextCursor::MoveAnchor, 1);
+	if (!nextBlockExists) {
+		m_shim.cursor.movePosition(QTextCursor::EndOfBlock, QTextCursor::MoveAnchor, 1);
+		m_shim.cursor.insertBlock();
+	}
+}
+
 void ScriptingTextBuffer::print(const QString& text) {
+	static const int tabStop = 4;
 	QMutexLocker locker(&m_mutex);
-	for (QString split : text.split('\n')) {
-		while (m_shim.cursor.positionInBlock() + split.length() > m_dims.width()) {
-			int cut = m_dims.width() - m_shim.cursor.positionInBlock();
-			if (!m_shim.cursor.atBlockEnd()) {
-				m_shim.cursor.movePosition(QTextCursor::EndOfBlock, QTextCursor::KeepAnchor);
-			}
-			m_shim.cursor.insertText(split.left(cut));
-			if (m_shim.cursor.atEnd()) {
-				m_shim.cursor.insertBlock();
+
+	QTextCursor& cursor = m_shim.cursor;
+	int lineWidth = m_dims.width();
+
+	for (const QChar& ch : text) {
+		if (ch == '\t') {
```

Comment:
Do we want to do this with a switch statement? (_Can_ we do this with a switch statement?)

Reply:
It COULD be done with a switch statement but I'm not sure it would make the code look any better.

Reply:
Well... After refactoring to fix the performance issue, maybe it would help.

Reply:
What do you think?